File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
export const COLORS = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white'
] as const

export type Color = typeof COLORS[number]

export function decodedValue(colors: Color[]): number {
  const first = String(COLORS.indexOf(colors[0]))
  const second = String(COLORS.indexOf(colors[1]))
  return Number(first + second)
}

File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.Objects;

class ResistorColorDuo {
    int value(String[] colors) {
        return Integer.parseInt(
                colorCode(colors[0]) + colorCode(colors[1])
        );
    }


    String[] colors = new String[]{"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    String colorCode(String color) {
        for (int i = 0; i < colors.length; i++) {
            if (Objects.equals(colors[i], color)) return String.valueOf(i);
        }
        return "" + -1;
    }
}
